include/zephyr/net/ieee802154.h

```diff
@@ -23,44 +25,183 @@ extern "C" {
+ * with this API.** It is of interest to subsystem maintainers only.
+ *
+ * The API implements and extends the following structures:
+ *    - implements Zephyr's internal L2-level socket and network context
+ *      abstractions (context/socket operations, see @ref net_l2),
+ *    - protocol-specific extension to the interface structure (see @ref net_if)
+ *    - protocol-specific extensions to the network packet structure
+ *      (see @ref net_pkt),
+ *
+ * @note All section, table and figure references are to the IEEE 802.15.4-2020
+ * standard.
+ *
  * @{
  */
 
-/* References are to the IEEE 802.15.4-2020 standard */
-#define IEEE802154_MAX_PHY_PACKET_SIZE	127 /* see section 11.3, aMaxPhyPacketSize */
-#define IEEE802154_FCS_LENGTH		2   /* see section 7.2.1.1 */
+#define IEEE802154_MAX_PHY_PACKET_SIZE	127 /**< see section 11.3, aMaxPhyPacketSize */
+#define IEEE802154_FCS_LENGTH		2   /**< see section 7.2.1.1 */
+
+/**
+ * @brief IEEE 802.15.4 "hardware" MTU (not to be confused with L3/IP MTU), i.e.
+ * the actual payload available to the next higher layer.
+ *
+ * @details This is equivalent to the IEEE 802.15.4 MAC frame length minus
+ * checksum bytes which is again equivalent to the PHY payload aka PSDU length
+ * minus checksum bytes. This definition exists for compatibility with the same
+ * concept in Linux and Zephyr's L3. It is not a concept from the IEEE 802.15.4
+ * standard.
+ *
+ * @note Currently we only support the original frame size from the 2006
+ * standard version and earlier. The 2015+ standard introduced PHYs with larger
+ * PHY payload. These are not (yet) supported in Zephyr.
+ */
 #define IEEE802154_MTU			(IEEE802154_MAX_PHY_PACKET_SIZE - IEEE802154_FCS_LENGTH)
+
 /* TODO: Support flexible MTU and FCS lengths for IEEE 802.15.4-2015ff */
 
+/** IEEE 802.15.4 short address length. */
 #define IEEE802154_SHORT_ADDR_LENGTH	2
+
+/** IEEE 802.15.4 extended address length. */
 #define IEEE802154_EXT_ADDR_LENGTH	8
+
+/** IEEE 802.15.4 maximum address length. */
 #define IEEE802154_MAX_ADDR_LENGTH	IEEE802154_EXT_ADDR_LENGTH
 
+/**
+ * A special channel value that symbolizes "all" channels or "any" channel -
+ * depending on context.
+ */
 #define IEEE802154_NO_CHANNEL		USHRT_MAX
 
-/* See IEEE 802.15.4-2020, sections 6.1 and 7.3.5 */
+/**
+ * @{
+ * See sections 6.1 and 7.3.5
+ */
 #define IEEE802154_BROADCAST_ADDRESS	     0xffff
 #define IEEE802154_NO_SHORT_ADDRESS_ASSIGNED 0xfffe
+/** @} */
 
-/* See IEEE 802.15.4-2020, section 6.1 */
+/* See section 6.1 */
 #define IEEE802154_BROADCAST_PAN_ID 0xffff
 
-/* See IEEE 802.15.4-2020, section 7.3.5 */
+/* See section 7.3.5 */
 #define IEEE802154_SHORT_ADDRESS_NOT_ASSOCIATED IEEE802154_BROADCAST_ADDRESS
 #define IEEE802154_PAN_ID_NOT_ASSOCIATED	IEEE802154_BROADCAST_PAN_ID
 
+/** interface-level security attributes, see section 9.5. */
 struct ieee802154_security_ctx {
+	/** section 9.5, secFrameCounter */
```

Comment:
Picking just one example but this applies to many similar occurrences: is this really how you want to document these? I guess the intention is to not duplicate any information that could be found in the cited reference, but that makes for an odd/unhelpful "brief" description of the symbol, no? Shouldn't it be along the lines of "Frame counter bla bla bla.\n Refer to section 9.5, secFrameCounter for additional information."

Reply:
Very good question.

The only thing we've explicitly decided to be mandatory so far (#61227) is the spec pointer, so that we prove that we rely on standard primitives everywhere. The pointer definitely has practical relevance because it's the hard part to find the right section in the spec. I use these pointers myself all the time.

Additionally copying over at least a one-sentence definition from the spec in general might add value, though. In some instances I do it where I cannot assume that a typical developer will know the definition (and that was more or less my criterium so far). In most instances I don't when I just point to well known concepts that people in the subsystem should have a notion of.

The frame counter is a good example to discuss this. I'd expect any developer in the subsystem to know what it is. But should we copy over at least some single-sentence definition everywhere? I honestly don't have a strong opinion on that now.

I'd be very much interested in what you @kartben and @rlubos / @jukkar actually prefer as I'm definitely open to whatever you feel is the best solution.

Reply:
IMHO I feel that the generated output might look a bit odd if we just have short ref to the spec. So I am inclined to agree with @kartben here. I am sure that for the expert the variable like frame_counter is obvious, I am just looking the generated doc and currently it looks a bit lonely there. See it yourself at
https://builds.zephyrproject.io/zephyr/pr/62503/docs/connectivity/networking/api/ieee802154.html#c.ieee802154_security_ctx.frame_counter

Reply:
@kartben @jukkar I wanted to add it to this PR but I underestimated Zephyr's speed once again. :-D I'll provide a one sentence definitions to all fields and concepts over the upcoming PRs. Thanks for your feedback there.

Reply:
mmh yeah why was this already merged? I guess we're missing something in GitHub UI to make it clear how long a PR has *actually*  been opened for review, since it hasn't even been 24 hours AFAICT.

Reply:
Guess it has to do with the closing 3.5 merge window? Or maybe just for the simple reason that the PR had two approvals and no request asked for officially. :-) Most certainly no deeper reason. In any case your comment is 100% valid has been confirmed by others which means I'll definitely take care of this and it will happen.